### PR TITLE
DGtalCatch documentation

### PR DESCRIPTION
**Pull request by dcoeurjo**
First version of the Dgtal-Catch documentation

### CODE REVIEW

**Comment by dcoeurjo** [comment]
@rolanddenis I've enabled templated test case in DGtal's catch version :smile:
Please have a look to the doc

**Comment by rolanddenis** [comment]
Thanks, nice feature ! Should I modify the Linearizer test file ?
However, it cannot be used with integer template parameters and it is missing cartesian product of template parameters :wink:

**Comment by dcoeurjo** [comment]
@rolanddenis up to you for Linearizer... 
:smiley:

**Comment by dcoeurjo** [comment]
Please, can someone check the doc ?

File: src/DGtal/doc/moduleCatch.dox [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+@page moduleCatch Unit test in DGtal using Catch 
+@writers David Coeurjolly
+
+[TOC]
+
+
+\section sectCatch1 Introduction
+
+[Catch](https://github.com/philsquared/Catch) Is a C++ framework to
+help unit test design. More precisely, thanks to couple of C++, Catch
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/doc/moduleCatch.dox]: "thanks to couple of C++" I do not understand what you mean.

**Comment by JacquesOlivierLachaud** [comment]
Thanks for the documentation and the template version of catch, which can be convenient !

**Comment by JacquesOlivierLachaud** [comment]
Passed all tests. So except for the remark above, it is mergable for me.

File: src/DGtal/doc/moduleCatch.dox (hunk 1 above)
Comment (dcoeurjo) [inline src/DGtal/doc/moduleCatch.dox]: "C++ macros" 
fixed.

**Comment by dcoeurjo** [comment]
Ok thanks, merging

**Pull request merged**